libraries/chain/db_witness_schedule.cpp

```diff
@@ -77,6 +77,22 @@ uint32_t database::get_slot_at_time(fc::time_point_sec when)const
    return (when - first_slot_time).to_seconds() / block_interval() + 1;
 }
 
+uint32_t database::count_missed_blocks( const signed_block& b )
```

Comment:
This function name is a bit misleading, it looks like a "getter" function, but actually it will update some data.

I think it's better to change the name to "update_witness_missed_blocks" or similar.